web/server/vue-cli/src/components/Report/ReportFilter/Filters/BaselineRunFilter.vue

```diff
@@ -1,40 +1,158 @@
 <template>
   <select-option
     :id="id"
-    title="Run Filter"
+    title="Run / Tag Filter"
     :bus="bus"
     :fetch-items="fetchItems"
     :selected-items="selectedItems"
     :search="search"
     :loading="loading"
+    :apply="apply"
+    @select="prevSelectedRuns = $event"
     @clear="clear(true)"
-    @input="setSelectedItems"
+    @on-menu-show="onMenuShow"
   >
-    <template v-slot:icon>
-      <v-icon color="grey">
-        mdi-play-circle
-      </v-icon>
+    <template v-slot:append-toolbar-title>
+      <selected-toolbar-title-items
+        v-if="selectedToolbarTitleItems.length"
+        :value="selectedToolbarTitleItems"
+      />
+    </template>
+
+    <template
+      v-slot:menu-content="{ items, prevSelectedItems, cancel, select }"
+    >
+      <v-tabs
+        v-model="tab"
+        fixed-tabs
+        hide-slider
+      >
+        <v-tab>Select Runs</v-tab>
+        <v-tab
+          :disabled="prevSelectedRuns.length === 0"
+          @click="specifyTagsTab"
+        >
+          Specify Tags
+          <v-tooltip max-width="200" right>
+            <template v-slot:activator="{ on }">
+              <v-icon
+                color="accent"
+                class="ml-1"
+                small
+                v-on="on"
+              >
+                mdi-help-circle
+              </v-icon>
+            </template>
+
+            <span>
+              Specify run tags of the selected runs to filter reports that
+              were <i>DETECTED BEFORE</i> the given date and
+              <i>NOT FIXED BEFORE</i> the given date.
```

Comment:
Because a tag is used here probably it would be nice to mention it, I mean something like this:
```
were <i>DETECTED BEFORE</i> and  <i>NOT FIXED BEFORE</i> the date when the selected tag was created .
```